.github/workflows/weekly-build.yml

```diff
@@ -0,0 +1,168 @@
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+env:
+  # renovate: datasource=go depName=github.com/florianl/bluebox
+  BLUEBOX_VERSION: v0.0.1
+
+jobs:
+  skip-check:
+    name: Skip check
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip-check.outputs.should_skip }}
+    permissions:
+      actions: write
+      contents: read
+    steps:
+      - id: skip-check
+        uses: fkirc/skip-duplicate-actions@f75f66ce1886f00957d99748a42c724f4330bdcf # v5.3.1
+        with:
+          do_not_skip: '["schedule", "workflow_dispatch"]'
+          paths: |-
+            [
+              "**.go",
+              ".github/workflows/build.yml",
+              ".go-version",
+              "3rdparty",
+              "Makefile",
+              "bpf/**",
+              "go.mod",
+              "go.sum",
+              "kerneltest/**",
+              "testdata/**"
+            ]
+          skip_after_successful_duplicate: false
+
+  vmtest:
+    name: kernel tests
+    runs-on: ubuntu-latest
+    needs: skip-check
+    if: ${{ needs.skip-check.outputs.should_skip != 'true' }}
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          go-version-file: .go-version
+
+      - name: Set up Clang
+        uses: KyleMayes/install-llvm-action@be40c5af3a4adc3e4a03199995ab73aa37536712 # v1.9.0
+        with:
+          version: "17.0.6"
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -y
+          sudo apt -y install qemu-system-x86 curl
+          sudo apt-get install -yq libelf-dev zlib1g-dev
+
+      - name: Initialize and update git submodules
+        run: git submodule init && git submodule update
+
+      - name: Build libbpf
+        run: make libbpf
+
+      - name: Build initramfs
+        run: |
+          go install "github.com/florianl/bluebox@${BLUEBOX_VERSION}"
+          make initramfs
+
+      - name: Run vmtests
+        run: ./kerneltest/vmtest.sh
+
+  go-build-test:
+    name: Go Build
+    runs-on: ubuntu-latest
+    needs: skip-check
+    if: ${{ needs.skip-check.outputs.should_skip != 'true' }}
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          go-version-file: .go-version
+
+      # Installs gofumpt, golangci-lint, and govulncheck.
+      - name: Set up environment
+        run: ./env.sh
+
+      - name: Set up Clang
+        uses: KyleMayes/install-llvm-action@be40c5af3a4adc3e4a03199995ab73aa37536712 # v1.9.0
+        with:
+          version: "17.0.6"
```

Comment:
Maybe we should use https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs and test against several versions.

Reply:
sounds good; since this GH action will run once a week, we can manage to test on multiple versions

In [build.yml](https://github.com/parca-dev/parca-agent/blob/main/.github/workflows/build.yml#L109-L112) we regularly test on clang version `14`, what versions would you recommend for this new weekly action, perhaps, [`15.0.7`, `16.0.6`, `17.0.6`] which are the latest patch releases of major release versions. WDYT?